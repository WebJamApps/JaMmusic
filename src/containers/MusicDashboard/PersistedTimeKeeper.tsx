import React from 'react';
import TimeKeeper from 'react-timekeeper';

export interface IPkProps {
  time:string, setShowTime: (...args:any) => any, setFormTime: (...args:any) => any
}

export const PersistedTimeKeeper = (props:IPkProps) =>{
  return (
    <TimeKeeper
            time={props.time}
            switchToMinuteOnHourSelect
            closeOnMinuteSelect
            onDoneClick={() => { props.setShowTime(false); props.setFormTime(props.time); }}
            onChange={(data) => { props.setFormTime(data.formatted12); }}
          />
  );
};
