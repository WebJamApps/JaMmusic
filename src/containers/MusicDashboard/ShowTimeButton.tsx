
export function ShowTimeButton(props:{ setShowTime:(...args:any)=>any, time:string }):JSX.Element {
  return (
    <div className="time-div">
      Time selected:
      {' '}
      {props.time}
      <br />
      <button className="show-clock" type="button" onClick={() => props.setShowTime(true)}>Show Clock</button>
    </div>
  );
}
