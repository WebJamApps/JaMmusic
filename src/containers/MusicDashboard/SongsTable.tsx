import React from 'react';
import MUIDataTable from 'mui-datatables';
import { ISong, SongsContext } from '../../providers/Songs.provider';
import songsTableUtils from './songsTableUtils';

type Props = {
  sData?:ISong[]
};
const SongsTable:React.FC<Props> = ({ sData }): JSX.Element => {
  const { test, songs } = React.useContext(SongsContext);
  let tableData = sData !== undefined ? sData : songs;
  tableData = tableData.length > 0 ? songsTableUtils.addButtons(tableData) : [];
  // eslint-disable-next-line no-console
  console.log(test);
  return (
    <div style={{ padding: '10px', marginBottom: '-20px' }}>
      {tableData.length > 0 ? (
        <MUIDataTable
          options={{
            filterType: 'dropdown',
            pagination: false,
            responsive: 'standard',
            filter: false,
            download: false,
            search: false,
            print: false,
            viewColumns: false,
            selectableRows: 'none',
            fixedHeader: false,
          }}
          columns={['modify', 'title', 'url', 'artist', 'category', 'album', 'image', 'composer', 'year']}
          data={tableData}
          title=""
        />
      ) : null }
    </div>
  );
};
export default SongsTable;
