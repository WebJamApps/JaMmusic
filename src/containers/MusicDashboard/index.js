import React, { Component, useState } from 'react';
import moment from 'moment';
import { withRouter, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DateFnsUtils from "@date-io/date-fns";
import {
  DatePicker,
  MuiPickersUtilsProvider,
  KeyboardTimePicker,
} from '@material-ui/pickers';
import mapStoreToProps from '../../redux/mapStoreToProps';

export class MusicDashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      redirect: false,
      date: '',
      time: '',
      tickets: '',
      more: '',
      selectedDate: new Date()
    };
    this.onChange = this.onChange.bind(this);
    this.createTour = this.createTour.bind(this);
    this.validateForm = this.validateForm.bind(this);
    this.createTourApi = this.createTourApi.bind(this);
    this.handleDateChange = this.handleDateChange.bind(this)
  }

  componentDidMount() { document.title = 'Music Dashboard | Web Jam LLC'; }

  handleDateChange(e) {
    this.setState({ selectedDate: e })
    console.log(e)
  }
  onChange(evt) {
    evt.preventDefault();
    this.setState({ [evt.target.id]: evt.target.value });
  }

  validateForm() {
    const {
      date, time, location, venue,
    } = this.state;
    if (date && time && location && venue && date !== '') return false;
    return true;
  }

  createTourApi(tour1) { // eslint-disable-line class-methods-use-this
    const { scc, auth } = this.props;
    const tour = tour1;
    tour.datetime = new Date(tour.date);
    const m = moment(tour.date, 'YYYY-MM-DD');
    tour.date = m.format('ll');
    scc.emit('newTour', { tour, token: auth.token });
    this.setState({ redirect: true });
  }

  createTour() {
    const {
      date, time, location, venue, tickets, more,
    } = this.state;
    const tour = {
      date, time, location, venue, tickets, more,
    };
    return this.createTourApi(tour);
  }

  makeInput(type, name, isRequired, onChange, value) { // eslint-disable-line class-methods-use-this
    return (
      <label htmlFor={name} style={{ marginTop: '14px', paddingTop: 0 }}>
        {isRequired ? '* ' : ''}
        {name[0].toUpperCase() + name.slice(1)}
        <br />
        <input id={name} type={type} name={name} onChange={onChange} required={isRequired} value={value} />
      </label>
    );
  }

  newTourForm(date, time, buttonStyle) {
    const {
      venue, location, tickets, more,
    } = this.state;
    return (
      <form id="new-tour" style={{ marginTop: '4px', paddingLeft: '10px' }}>
        {this.makeInput('date', 'date', true, this.onChange, date)}
        {this.makeInput('text', 'time', true, this.onChange, time)}
        {this.makeInput('text', 'venue', true, this.onChange, venue)}
        {this.makeInput('text', 'location', true, this.onChange, location)}
        {this.makeInput('text', 'tickets', false, this.onChange, tickets)}
        {this.makeInput('text', 'more', false, this.onChange, more)}
        <div  className = "divStyle">
          <MuiPickersUtilsProvider utils={DateFnsUtils} >
            <KeyboardTimePicker
              id="time-picker"
              label="Time picker"
              value={this.selectedDate}
              onChange={this.handleDateChange}
              KeyboardButtonProps={{ 'aria-label': 'change time' }}
            />
          </MuiPickersUtilsProvider>
        </div>
        <div style={{ textAlign: 'right', marginTop: '10px', maxWidth: '85%' }}>
          <span style={{ fontSize: '16px', marginRight: '38%', fontFamily: 'Habibi' }}><i>* Required</i></span>
          <button style={buttonStyle} disabled={this.validateForm()} type="button" onClick={this.createTour}>Create Tour</button>
        </div>
      </form>
    );
  }


  render() {
    const {
      date, time, buttonStyle, redirect,
    } = this.state;
    return (
      <div className="page-content" >
        {redirect ? <Redirect to="/music" /> : null}
        <h3 style={{ textAlign: 'center', margin: '14px', fontWeight: 'bold' }}>Music Dashboard</h3>
        <h5 style={{ textAlign: 'center', marginBottom: 0 }}>Create a New Tour</h5>
        {this.newTourForm(date, time, buttonStyle)}
      </div>
    );
  }
}
MusicDashboard.propTypes = { scc: PropTypes.shape({ emit: PropTypes.func }), auth: PropTypes.shape({ token: PropTypes.string }).isRequired };
MusicDashboard.defaultProps = { scc: { emit: () => { } } };
export default withRouter(connect(mapStoreToProps)(MusicDashboard));
