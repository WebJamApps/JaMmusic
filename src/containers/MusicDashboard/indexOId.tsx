import React, { Component } from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { connect } from 'react-redux';
import type { AGClientSocket } from 'socketcluster-client';
import type { Dispatch, AnyAction } from 'redux';
import mapStoreToProps, { Iimage, Auth } from '../../redux/mapStoreToProps';
import Forms from '../../lib/forms';
import commonUtils from '../../lib/commonUtils';
// import Controller, { MusicDashboardController } from './MusicDashboardController';
import { TourEditor } from '../../components/TourEditor';
import { DashNavigationButtons } from '../../components/DashNavigationButtons';

interface MusicDashboardProps extends RouteComponentProps<Record<string, string | undefined>> {
  dispatch: Dispatch<AnyAction>;
  scc: AGClientSocket;
  auth: Auth;
  editPic: Iimage;
  images: Iimage[];
}

type MusicDashboardState = {
  title: string,
  url: string,
  showCaption: string,
  location: string;
  venue: string;
  redirect: boolean;
  date: string;
  datetime: string;
  time: Date | null;
  tickets: string;
  more: string;
  [x: number]: number;
  navState:{ navSong:boolean, navPhoto:boolean, navTour: boolean };
};

const InitialState = {
  songState: {
    image: '', composer: '', year: new Date().getFullYear(), album: '', title: '', url: '', artist: '', category: 'original', _id: '',
  },
  datetime: '',
  title: '',
  url: '',
  showCaption: '',
  redirect: false,
  date: '',
  time: new Date(),
  tickets: '',
  more: '',
  venue: '',
  location: '',
  navState: { navSong: true, navPhoto: false, navTour: false },
};
export class MusicDashboard extends Component<MusicDashboardProps, MusicDashboardState> {
  commonUtils: { setTitleAndScroll: (pageTitle: string, width: number) => void };

  // controller: MusicDashboardController;

  public forms = Forms;

  constructor(props: MusicDashboardProps) {
    super(props);
    this.state = InitialState;
    // this.controller = new Controller(this);
    this.commonUtils = commonUtils;
    this.handleNavClick = this.handleNavClick.bind(this);
  }

  componentDidMount(): void { this.commonUtils.setTitleAndScroll('Music Dashboard', window.screen.width); }

  // eslint-disable-next-line react/sort-comp
  handleNavClick(e: React.MouseEvent<HTMLButtonElement>): void {
    if (e.currentTarget.id === 'Songs-Button') {
      this.setState({ navState: { navSong: true, navPhoto: false, navTour: false } });
    }
    if (e.currentTarget.id === 'Gigs-Button') {
      this.setState({ navState: { navSong: false, navPhoto: false, navTour: true } });
    }
    if (e.currentTarget.id === 'Photos-Button') {
      this.setState({ navState: { navSong: false, navPhoto: true, navTour: false } });
    }
  }

  render(): JSX.Element {
    const { navState } = this.state;
    return (
      <div className="page-content">
        <DashNavigationButtons comp={this} />
        {/* {navState.navSong ? (this.controller.songBlock()) : null}
        {navState.navPhoto ? (this.controller.pictureBlock()) : null} */}
        {/* {navState.navTour ? <TourEditor /> : null} */}
      </div>
    );
  }
}

export default withRouter(connect(mapStoreToProps, null)(MusicDashboard));
