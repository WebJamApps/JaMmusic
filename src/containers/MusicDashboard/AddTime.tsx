
import createPersistedState from 'use-persisted-state';
import { PersistedTimeKeeper } from './PersistedTimeKeeper';
import { ShowTimeButton } from './ShowTimeButton';
const useTimeState = createPersistedState('showTime', sessionStorage);
type Props = {
  setFormTime: (args0: string) => void, initTime: string, show:boolean
};
export default function AddTime({ setFormTime, initTime, show }: Props): JSX.Element { // eslint-disable-line react/prop-types
  const time = initTime !== '' ? initTime : '12:34pm';
  const [showTime, setShowTime] = useTimeState(show);
  return (
    <div className="clock-input">
      <br />
      <span>* Time</span>
      <br />
      {showTime
        ? (
          <PersistedTimeKeeper time={time} setShowTime={setShowTime} setFormTime={setFormTime}/>
        ) : null}
      {!showTime
        ? (
          <ShowTimeButton setShowTime={setShowTime} time={time}/>
        ) : null}
    </div>
  );
}
