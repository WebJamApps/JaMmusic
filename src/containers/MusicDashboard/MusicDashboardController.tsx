import React from 'react';
import Superagent from 'superagent';
import type { MusicDashboard } from './index';
import Forms from '../../lib/forms';
import SongsTable from '../../components/SongsTable';
import SongEditorUtils from '../../components/SongEditor/songEditorUtils';
import SongEditor from '../../components/SongEditor';
import { PicEditor } from '../../components/PicEditor';

export class MusicDashboardController {
  view: MusicDashboard;

  forms = Forms;

  superagent = Superagent;

  songEditorUtils = SongEditorUtils;

  constructor(view: MusicDashboard) {
    this.view = view;
    this.changePicDiv = this.changePicDiv.bind(this);
    this.addPic = this.addPic.bind(this);
    this.modifySongsSection = this.modifySongsSection.bind(this);
  }

  // eslint-disable-next-line class-methods-use-this
  modifySongsSection():JSX.Element {
    const { auth, dispatch } = this.view.props;
    return (
      <div
        className="search-table-outer"
        style={{
          maxWidth: '96%', margin: 'auto', zIndex: 0,
        }}
      >
        <h5 style={{ textAlign: 'center', marginBottom: '3px' }}>Modify Songs</h5>
        <SongsTable token={auth.token} dispatch={dispatch} />
      </div>
    );
  }

  addPic(): void {
    const { picTitle, picUrl, showCaption } = this.view.state;
    const { scc, auth } = this.view.props;
    const image = {
      title: picTitle,
      url: picUrl,
      caption: showCaption,
      type: 'JaMmusic-music',
    };
    scc.transmit('newImage', { image, token: auth.token });
    window.location.assign('/music');
  }

  changePicDiv(): JSX.Element {
    let { editPic } = this.view.props;
    const { picTitle, picUrl, showCaption } = this.view.state;
    if (!editPic) {
      editPic = {
        title: '',
        url: '',
        type: '',
        caption: '',
      };
    }
    return (
      <div
        className="material-content elevation3"
        style={{ maxWidth: '320px', margin: '30px auto' }}
      >
        <h5 style={{ marginBottom: 0 }}>
          {editPic && editPic._id ? 'Edit ' : 'Add '}
          Pictures
        </h5>
        <PicEditor comp={this.view} controller={this} editPic={editPic} />
      </div>
    );
  }

  pictureBlock(): JSX.Element {
    return (
      <div className="material-content elevation3" style={{ maxWidth: '9.1in', margin: 'auto' }}>
        <h5 style={{ textAlign: 'center', marginBottom: 0 }}>Modify Photo Slideshow</h5>
        {this.changePicDiv()}
      </div>
    );
  }

  songBlock(): JSX.Element {
    return (
      <div className="Song-Block">
        <p>&nbsp;</p>
        {/* {this.changeSongDiv()} */}
        <SongEditor controller={this} editSong={this.view.props.editSong} songState={this.view.state.songState} comp={this.view} />
        <p>&nbsp;</p>
        {this.modifySongsSection()}
        <p>&nbsp;</p>
      </div>
    );
  }
}
export default MusicDashboardController;
