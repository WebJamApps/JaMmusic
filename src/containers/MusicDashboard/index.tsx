import React, { Component } from 'react';
import { Editor } from '@tinymce/tinymce-react';
import moment from 'moment';
import { withRouter, Redirect, RouteComponentProps } from 'react-router-dom';
import { connect } from 'react-redux';
import mapStoreToProps from '../../redux/mapStoreToProps';
import forms from '../../lib/forms';
import commonUtils from '../../lib/commonUtils';
import AddTime from '../../lib/timeKeeper';
import Ttable from '../../components/tour-table';

interface MusicDashboardProps extends RouteComponentProps<any> {
  dispatch: (...args: any) => any;
  scc: { transmit: (...args: any[]) => any };
  auth: { token: string };
  editTour: { date?: string; time?: string; tickets?: string; more?: string; venue?: string; location?: string; _id?: string; datetime?: string };
}
type MusicDashboardState = {
  location: string;
  venue: any | string;
  redirect: boolean;
  date: string;
  time: string;
  tickets: string;
  more: string;
  [x: number]: any;
};
export class MusicDashboard extends Component<MusicDashboardProps, MusicDashboardState> {
  forms: any;

  commonUtils: any;

  constructor(props: any) {
    super(props);
    this.state = {
      redirect: false, date: '', time: '', tickets: '', more: '', venue: '', location: '',
    };
    this.forms = forms;
    this.onChange = this.onChange.bind(this);
    this.createTour = this.createTour.bind(this);
    this.validateForm = this.validateForm.bind(this);
    this.createTourApi = this.createTourApi.bind(this);
    this.handleEditorChange = this.handleEditorChange.bind(this);
    this.setFormTime = this.setFormTime.bind(this);
    this.commonUtils = commonUtils;
    this.checkEdit = this.checkEdit.bind(this);
    this.editTourAPI = this.editTourAPI.bind(this);
    this.resetEditForm = this.resetEditForm.bind(this);
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('Music Dashboard'); }

  onChange(evt: any) {
    const { editTour } = this.props;
    if (editTour.venue !== undefined) this.checkEdit();
    this.setState({ [evt.target.id]: evt.target.value });
  }

  setFormTime(data: any) {
    this.setState({ time: data });
  }

  checkEdit() {
    let {
      date, time, tickets, more, venue, location,
    } = this.state;
    const { editTour, dispatch } = this.props;
    if (date === '' && editTour.date !== undefined) { date = editTour.datetime.split('T')[0]; }//eslint-disable-line
    if (time === '' && editTour.time !== undefined) { time = editTour.time; }
    if (tickets === '' && editTour.tickets !== undefined) { tickets = editTour.tickets; }
    if (more === '' && editTour.more !== undefined) { more = editTour.more; }
    if (venue === '' && editTour.venue !== undefined) { venue = editTour.venue; }
    if (location === '' && editTour.location !== undefined) { location = editTour.location; }
    this.setState({
      date, time, tickets, more, venue, location,
    });

    if (editTour.venue !== undefined) {
      dispatch({ type: 'EDIT_TOUR', data: { _id: editTour._id } });
    }
  }

  resetEditForm(evt: { preventDefault: (...args: any) => any }) {
    evt.preventDefault();
    const { dispatch } = this.props;
    dispatch({ type: 'EDIT_TOUR', data: {} });
    this.setState({
      date: '', time: '', tickets: '', more: '', venue: '', location: '',
    });
  }

  handleEditorChange(venue: any) { this.setState({ venue }); }

  validateForm() {
    const {
      date, time, location, venue,
    } = this.state;
    if (date && time && location && venue && date !== '') return false;
    return true;
  }

  createTourApi(tour1: any) {
    const { scc, auth } = this.props;
    const tour = tour1;
    tour.datetime = tour.date;
    const m = moment(tour.date, 'YYYY-MM-DD');
    tour.date = m.format('ll');
    scc.transmit('newTour', { tour, token: auth.token });
    this.setState({ redirect: true });
    return true;
  }

  createTour() {
    const {
      date, time, location, venue, tickets, more,
    } = this.state;
    const tour = {
      date, time, location, venue, tickets, more,
    };
    return this.createTourApi(tour);
  }

  editor(venue: string) {
    return (
      <Editor
        value={venue}
        apiKey={process.env.TINY_KEY}
        init={{
          height: 400,
          menubar: 'insert',
          selector: 'textarea',
          menu: { format: { title: 'Format', items: 'forecolor backcolor' } },
          plugins: [
            'advlist autolink lists link image charmap print preview anchor',
            'searchreplace visualblocks code fullscreen',
            'insertdatetime media table paste code help wordcount',
          ],
          toolbar:
            'undo redo | formatselect | bold italic backcolor forecolor |'
            + 'alignleft aligncenter alignright alignjustify |'
            + 'bullist numlist outdent indent | removeformat | help',
        }}
        onEditorChange={this.handleEditorChange}
      />
    );
  }

  editTourAPI() {
    const {
      date, time, location, venue, tickets, more,
    } = this.state;
    const { editTour, scc, auth } = this.props;
    const tour = {
      date, time, location, venue, tickets, more, datetime: date,
    };
    const m = moment(tour.date, 'YYYY-MM-DD');
    tour.date = m.format('ll');
    scc.transmit('editTour', { tour, token: auth.token, tourId: editTour._id });
    this.resetEditForm({ preventDefault: () => { } });
    this.setState({ redirect: true });
    return true;
  }

  tourButtons() {
    const { editTour } = this.props;
    return (
      <div style={{ textAlign: 'left', marginTop: '10px', maxWidth: '85%' }}>
        <span style={{
          fontSize: '16px', marginRight: '20px', fontFamily: 'Habibi', position: 'relative', display: 'inline-block',
        }}
        >
          <i>* Required</i>
        </span>
        {editTour._id ? (
          <button className="floatRight" type="button" id="cancel-edit-pic" onClick={this.resetEditForm}>
            Cancel
          </button>
        ) : null}
        <button
          className="floatRight"
          disabled={this.validateForm()}
          type="button"
          onClick={editTour._id ? this.editTourAPI : this.createTour}
        >
          {editTour._id ? 'Edit' : 'Create'}
          {' '}
          Tour
        </button>
      </div>
    );
  }

  newTourForm() {
    let {
      location, tickets, more, date, time, venue,
    } = this.state;
    const { editTour } = this.props;
    if (editTour.date !== undefined) { date = editTour.datetime.split('T')[0]; }//eslint-disable-line
    if (editTour.time !== undefined) { time = editTour.time; }
    if (editTour.tickets !== undefined) { tickets = editTour.tickets; }
    if (editTour.more !== undefined) { more = editTour.more; }
    if (editTour.venue !== undefined) { venue = editTour.venue; }
    if (editTour.location !== undefined) { location = editTour.location; }
    return (
      <form id="new-tour" style={{ marginLeft: '4px', marginTop: '4px' }}>
        {this.forms.makeInput('date', 'Date', true, this.onChange, date)}
        <AddTime setFormTime={this.setFormTime} initTime={time} />
        <div className="horiz-scroll">
          <div style={{ width: '850px', margin: 'auto' }}>
            <p style={{ marginBottom: 0 }}>* Venue</p>
            {this.editor(venue)}
          </div>
        </div>

        {this.forms.makeInput('text', 'Location', true, this.onChange, location)}
        {this.forms.makeInput('text', 'Tickets', false, this.onChange, tickets)}
        {this.forms.makeInput('text', 'More', false, this.onChange, more)}
        {this.tourButtons()}
      </form>
    );
  }

  render() {
    const { redirect } = this.state;
    const { editTour } = this.props;
    return (
      <div className="page-content">
        {redirect ? <Redirect to="/music" /> : null}
        <h3 style={{ textAlign: 'center', margin: '14px', fontWeight: 'bold' }}>Music Dashboard</h3>
        <div className="material-content elevation3" style={{ maxWidth: '9.1in', margin: 'auto' }}>
          <h5 style={{ textAlign: 'center', marginBottom: 0 }}>
            {editTour._id ? 'Edit' : 'Create a New'}
            {' '}
            Tour Event
          </h5>
          {this.newTourForm()}
        </div>
        <p>&nbsp;</p>
        {!editTour._id ? (
          <div className="material-content elevation3" style={{ maxWidth: '10in', margin: 'auto' }}>
            <h5 style={{ textAlign: 'center', marginBottom: 0 }}>Modify Event</h5>
            <Ttable deleteButton />
          </div>
        ) : null}
        <p>&nbsp;</p>
        <p>&nbsp;</p>
      </div>
    );
  }
}

export default withRouter(connect(mapStoreToProps, null)(MusicDashboard));

