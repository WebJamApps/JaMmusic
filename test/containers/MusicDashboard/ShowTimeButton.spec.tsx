/* eslint-disable @typescript-eslint/no-explicit-any */
import * as React from 'react';
import { shallow } from 'enzyme';
import { ShowTimeButton } from '../../../src/containers/MusicDashboard/ShowTimeButton';

describe('ShowTimeButton function', () => {
  it('handles click to show the time widget', () => {
    const setShowTime = jest.fn();
    const wrapper = shallow(<ShowTimeButton time='' setShowTime={setShowTime}/>);
    expect(wrapper).toMatchSnapshot(); 
    wrapper.find('button.show-clock').simulate('click');
    expect(setShowTime).toHaveBeenCalled();
  });
});
