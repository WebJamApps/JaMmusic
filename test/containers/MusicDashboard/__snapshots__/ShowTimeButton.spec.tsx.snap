// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowTimeButton function handles click to show the time widget 1`] = `ShallowWrapper {}`;
